src/extension.ts

```diff
@@ -42,6 +44,21 @@ export function activate(context: vscode.ExtensionContext) {
       }
     }
   );
+  
+  //register separate stack report command and record actions
+  for (const action in commandMapping) {
+    if (Object.prototype.hasOwnProperty.call(commandMapping, action)) {
+      const command = commandMapping[action];
+      context.subscriptions.push(
+        vscode.commands.registerCommand(command,
+          (uri : vscode.Uri)=>{
+            record(ActionName[action]);
+            const fileUri = uri ? uri : vscode.window.activeTextEditor.document.uri;
+            multimanifestmodule.dependencyAnalyticsReportFlow(context, fileUri);
+      }))
+    }
+  }
+

```

Comment:
Thought of another easy approach, which doesn't need intermediate mapping between command and action.
```
const invokeFullStackReport = (uri : vscode.Uri) => {
     const fileUri = uri || vscode.window.activeTextEditor.document.uri;
     multimanifestmodule.dependencyAnalyticsReportFlow(context, fileUri);
};
   const recordAndInvoke = (origin: string, uri : vscode.Uri) => {
     record(origin);
     invokeFullStackReport(uri);
   }
    const commands = [ vscode.commands.registerCommand(TRIGGER_FULL_STACK_ANALYSIS_FROM_EDITOR, recordAndInvoke.bind(ActionName.vulnerabilityReportEditor)),
vscode.commands.registerCommand(TRIGGER_FULL_STACK_ANALYSIS_FROM_PIE_BTN, recordAndInvoke.bind(ActionName.vulnerabilityReportPieBtn)),
...
];
context.subscriptions.push(...commands);
```

Reply:
ya this looks pretty straight foward. 
let me try and update PR

Reply:
done PTAL